cluster-location-metadata/README.md

```diff
@@ -0,0 +1,343 @@
+---
+creation_date: 2025-01-30
+issues: []
+owners:
+- https://github.com/orgs/giantswarm/teams/team-honeybadger
+state: review
+summary: The document discusses the need for consistent location metadata for clusters, addressing customer communication and management needs. It proposes a unified location identifier metadata system on the cluster level, for cloud and on-premises clusters.
+---
+
+# Location metadata for clusters
+
+### Problem statement
+
+We have anecdotal evidence that customers think about our installations in terms of their physical location. They communicate in terms like these (paraphrased):
+
+- the Dublin management cluster
+- the dev cluster in Singapore
+- the China installation
+
+Some on-prem customers use location identifiers like city names in cluster descriptions.
+
+Relevant user stories in this context could be, to name two examples:
+
+- As a platform admin, I want to check the Kubernetes version of all clusters in a certain location.
+- As a service owner, I want to check the deployment status of certain apps in a certain location.
+
+With Cluster API on cloud providers we usually have location indicators in cluster resources, in various places. Each cloud provider uses their own identifier system, however they all make use of a string identifier. The proposed solution in this RFC would help simplify and unify location lookup for clusters by clients throughout cloud providers.
+
+In on-premises installations, no location information is currently available in Kubernetes resources. In metrics (PrometheusMimir) we annotate many series with a `region` label. However, in on-prem installations, the value for this label is always `onprem`. This RFC attempts to change that.
+
+### Key assumptions
+
+This RFC assumes that each cluster exists in one location only, and nodes of the same cluster are not distributed geographically.
+
+### Decision maker
+
+- Team Honeybadger as driver
+- KaaS teams, especially Rocket
+
+### Deadline
+
+None
+
+### Who is affected / stakeholders
+
+- Customers
+- Giant Swarm engineers
+
+### Preferred solution
+
+The solution proposed here consists of two parts:
+
+- Setting consistent metadata on clusters
+- Providing a default location ID system for on-prem customers
+- Setting the `region` label in metrics accordingly
+
+#### Consistent metadata on clusters
+
+We should set the [`topology.kubernetes.io/region`](https://kubernetes.io/docs/reference/labels-annotations-taints/#topologykubernetesioregion) label on the main cluster resource of all clusters. Currently (as of February 2025) this resource is of kind `Cluster`, group=`cluster.x-k8s.io/v1beta1`.
```

Comment:
Was this not supposed to be a node label used for scheduling? I am not sure if the topology label makes sense here

Reply:
The linked docs page states that it's used on Node, PersistentVolume. So far it seems it's not used on Cluster.

If we are uncertain about this, we can decide for a custom (giantswarm.io) label instead.

Reply:
I would rather we use well known labels tbh, i'm wondering if there are not some in the CAPI or kcp work that would fit better but it's still okay as is